local_planner/src/nodes/planner_functions.cpp

```diff
@@ -308,13 +308,56 @@ void getCostMatrix(const Histogram& histogram, const Eigen::Vector3f& goal,
     }
   }
 
-  float smoothing_margin_degrees = 45;
+  float smoothing_margin_degrees = 30;
   unsigned int smooth_radius = ceil(smoothing_margin_degrees / ALPHA_RES);
   smoothPolarMatrix(distance_matrix, smooth_radius);
 
+  generateCostImage(cost_matrix, distance_matrix, image);
   cost_matrix = cost_matrix + distance_matrix;
 }
 
+void generateCostImage(const Eigen::MatrixXf& cost_matrix,
+                       const Eigen::MatrixXf& distance_matrix,
+                       sensor_msgs::Image& image) {
+  float max_val = std::max(cost_matrix.maxCoeff(), distance_matrix.maxCoeff());
+  image.height = GRID_LENGTH_E;
+  image.width = GRID_LENGTH_Z;
+  image.encoding = "rgb8";
+  image.is_bigendian = 0;
+  image.step = 3 * image.width;
+
+  int image_size = static_cast<int>(image.height * image.width);
+  image.data.clear();
+  image.data.reserve(3 * image_size);
+
+  for (int e = GRID_LENGTH_E - 1; e >= 0; e--) {
+    for (int z = 0; z < GRID_LENGTH_Z; z++) {
+      float distance_cost = 255.f * distance_matrix(e, z) / max_val;
+      float other_cost = 255.f * cost_matrix(e, z) / max_val;
+      image.data.push_back((int)std::max(0.0f, std::min(255.f, distance_cost)));
```

Comment:
`static_cast<int>`

Reply:
Now that I think about it, shouldn't it be a byte? `sensor_msgs::Image` has a uint8...